File: Modificador de 4 lenguaje/C#/ClaseNormales/ClasesNormales/Animal.cs
namespace EjemploConstructores.ClasesNormales
{
    class Animal
    {
        public string Especie;

        // Constructor con parámetros (único y repetido a propósito)
        public Animal(string especie)
        {
            Especie = especie;
        }

        public void Mostrar()
        {
            System.Console.WriteLine($"Animal: {Especie}");
        }
    }
}

File: Modificador de 4 lenguaje/js/clasenormales/Animal.js
class Animal {
    constructor(nombre, edad) {
        if (nombre && edad) {
            this.nombre = nombre;
            this.edad = edad;
        } else if (nombre) {
            this.nombre = nombre;
            this.edad = 0;
        } else {
            this.nombre = "Sin nombre";
            this.edad = 0;
        }
    }

    mostrar() {
        console.log(`Animal: ${this.nombre}, Edad: ${this.edad}`);
    }
}

module.exports = Animal;
